packages/react-core/src/hooks/useGetUrl.ts

```diff
@@ -19,13 +24,17 @@ const INIT_STATE: UseGetUrlOutput = {
   isLoading: true,
 };
 
-const useGetUrl = (input: UseGetUrlInput): UseGetUrlOutput => {
+type GetUrl = (
+  input: GetUrlInput | GetUrlWithPathInput
+) => Promise<GetUrlOutput>;
+
+export default function useGetUrl(input: UseGetUrlInput): UseGetUrlOutput {
   const [result, setResult] = React.useState(() => INIT_STATE);
   React.useEffect(() => {
     const { onError, ...getUrlInput } = input;
     let ignore = false;
 
-    getUrl(getUrlInput)
+    (getUrl as GetUrl)(getUrlInput)
```

Comment:
No way to avoid this assertion?

Reply:
IMO adding a type predicate because the overload is not exposing the base definition is just extra overhead